cmd/function-sidecar.go

```diff
@@ -115,69 +112,12 @@ func main() {
 		defer d.Close()
 	}
 
-	go func() {
-		for {
-			select {
-			// Incoming message
-			case msg, open := <-consumer.Messages():
-				if open {
-					messageIn, err := wireformat.FromKafka(msg)
-					log.Printf(">>> %s\n", messageIn)
-					if err != nil {
-						log.Printf("Error receiving message from Kafka: %v", err)
-						break
-					}
-					dispatcher.Input() <- messageIn
-					consumer.MarkOffset(msg, "") // mark message as processed
-				} else {
-					// Kafka closed
-					log.Print("Exiting Kafka Consumer loop")
-					return
-				}
-			}
-		}
-	}()
-
-	go func() {
-		for {
-			select {
-			// Result message
-			case resultMsg, open := <-dispatcher.Output(): // Make sure to drain channel even if output==""
-				if open {
-					if output != "" {
-						log.Printf("<<< %s\n", resultMsg)
-						producerMessage, err := wireformat.ToKafka(resultMsg)
-						if err != nil {
-							log.Printf("Error encoding message: %v", err)
-							break
-						}
-						producerMessage.Topic = output
-						select {
-						case producer.Input() <- producerMessage:
-						}
-					} else {
-						log.Printf("=== Not sending function return value as function did not provide an output channel. Raw result = %s\n", resultMsg)
-					}
-				} else {
-					log.Print("Exiting Kafka Producer loop")
-					return
-				}
-			}
-		}
-	}()
+	carrier.Run(consumer, producer, dispatcher, output)
```

Comment:
Do you think you can refactor the test in `function-sidecar_test.go` to use this and collaborators?
I really don't like the fact that it runs a compiled binary (that must exist) -- this will hinder the switch to the new Docker scheme, for one thing.

Or we could do this as a separate PR, of course

Reply:
@ericbottard I think that's worth trying, but in a separate PR. I raised [issue 45](https://github.com/projectriff/function-sidecar/issues/45) to cover this.